arpav_ppcv/webapp/v2/routers/observations.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from typing import Annotated
+
+import pydantic
+from fastapi import (
+    APIRouter,
+    Depends,
+    Request,
+)
+from sqlmodel import Session
+
+from .... import database
+from ... import dependencies
+from ..schemas import observations
+
+logger = logging.getLogger(__name__)
+router = APIRouter()
+
+
+@router.get("/stations/", response_model=observations.StationList)
```

Comment:
No, I think it is correct. Using a trailing slash for API endpoints that return a list of records is a standard pattern. 

We then have `/stations/{station_id}` (without trailing slash) for providing individual records.

Reply:
I will make the suggested changes then, no worries :wink: 

Just to get a little nerdy here, I usually refer back to the really nice [RESTful Web APIs](http://restfulwebapis.com/) book when I need a refresher of these types of concepts - coincidentally, the author shares the same opinion as me on the advice for designing collection endpoint URLs :stuck_out_tongue: 

This is an excerpt from the book about this topic:

> **URL Design Doesn’t Matter**
> 
> Some API design guides, including the original RESTful Web Services, spend a lot of time talking about the URLs you should assign to your resources. Each URL you serve should clearly identify the resource in such a way that a human being looking at the URL can figure out what’s on the other end.If you publish a resource that’s a collection of user accounts, it should be called something like `/users/`. Subordinate resources should be published beneath their parents. So the resource representing Alice’s account should be given a URL like `/users/alice`. Well, sort of.
> 
> Technically speaking, none of this matters. A URL is just the address of a resource, which a client can use to get a representation. The URL doesn’t technically say anything about the resource or its representation. The Architecture of the World Wide Web, Volume One puts it this way:
> 
> > It is tempting to guess the nature of a resource by inspection of a URI that identifies it. However, the Web is designed so that agents communicate resource information state through representations, not identifiers. In general, one cannot determine the type of a resource representation by inspecting a URI for that resource.
> 
> (...)
> 
> When you look at a URL and try to make sense of the underlying resource, you’re trying to figure out the resource’s application-level semantics. That’s fine. The advice on URL design given in other books and tutorials is good advice. But I won’t rehash that advice in this book, because I don’t want your APIs relying on URLs to convey their application-level semantics. We have more reliable ways of describing that stuff: media type definitions and machine-readable profiles.

In the last sentence, machine readable profiles would refer to our OpenAPI document, in this particular case.

Anyway, as said I'm cool with changing the URLs - this was just some geeky context for my preference.


WRT the spectral tests, yes good idea - I'll add those later. Can you please open an issue for this?